snapcraft/internal/sources.py

```diff
@@ -80,6 +82,28 @@ def __init__(self, source, source_dir, source_tag=None,
         self.source_branch = source_branch
 
 
+class FileBase(Base):
+
+    def pull(self):
+        if common.isurl(self.source):
+            self.download()
+        else:
+            shutil.copy2(self.source, self.source_dir)
+
+        self.provision(self.source_dir)
```

Comment:
I've been told lately it is not really necessary, it just needs to behave and act like a duct :-)

I am only saying this since it is tightly controlled code.